Utils/DecryptBlobFile/README.md

```diff
@@ -0,0 +1,125 @@
+---
+page_type: sample
+languages:
+- csharp
+products:
+- azure
+- azure-search
+- azure-key-vault
+- azure-storage
+name: Decrypt blob file sample skill for cognitive search
+urlFragment: azure-decryptblob-sample
+description: This custom skill downloads and decrypts a file that was encrypted in Azure Blob Storage and returns it back to Azure Cognitive Search to be indexed.
+azureDeploy: https://raw.githubusercontent.com/Azure-Samples/azure-search-power-skills/master/Utils/DecryptBlobFile/azuredeploy.json
+---
+
+# DecryptBlobFile
+
+This custom skill downloads and decrypts a file that was encrypted in Azure Blob Storage and returns it back to Azure Cognitive Search to be processed and indexed. It is meant to be used in combination with the built-in [DocumentExtractionSkill](https://docs.microsoft.com/azure/search/cognitive-search-skill-document-extraction) to allow you to index encrypted files without needing to worry about them being stored unecrypted at rest. For more details on how to encrypt files in blob storage, [see this tutorial](https://docs.microsoft.com/azure/storage/blobs/storage-encrypt-decrypt-blobs-key-vault).
+
+## Requirements
+
+In addition to the common requirements described in [the root `README.md` file](../../README.md), this function requires key get access to the [Azure Key Vault](https://azure.microsoft.com/services/key-vault/) resource where the key that was used to encrypt the files stored in Azure Blob Storage lives. This access should be granted by [setting an access policy on the Key Vault](https://docs.microsoft.com/azure/key-vault/general/assign-access-policy-portal) with the principal being the Azure Function instance that the skill is deployed to.
+
+## Settings
+
+This function doesn't require any application settings.
+
+## Deployment
+
+[![Deploy to Azure](https://azuredeploy.net/deploybutton.svg)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure-Samples%2Fazure-search-power-skills%2Fmaster%2FUtils%2FDecryptBlobFile%2Fazuredeploy.json)
+
+## Sample Input:
+
+```json
+{
+    "values": [
+        {
+            "recordId": "record1",
+            "data": { 
+                "blobUrl": "http://blobStorage.com/myencryptedfile",
+                "sasToken": "?sas=123&otherSasInfo=456"
+            }
+        }
+    ]
+}
+```
+
+## Sample Output:
+
+```json
+{
+    "values": [
+        {
+            "recordId": "record1",
+            "data": {
+                "decrypted_file_data": {
+                    "$type": "file",
+                    "data": "<base64 encoded decrypted file data>"
+                }
+            },
+            "errors": null,
+            "warnings": null
+        }
+    ]
+}
+```
+
+## Sample Skillset Integration
+
```

Comment:
Is it a relevant scenario at all, and is it worth pointing out what to do if not all files are encrypted, and is there a way in that case to avoid the round-trip to the custom skill with a conditional or something similar?

Reply:
So I don't think this is relevant; even if it is possible that a container contains some encrypted blobs and some not, because since we are requesting the user set "dataToExtract": "storageMetadata", they will need this skill and the document extraction skill to get the content for the unencrypted blobs as well.  I should validate that this skill is able to download blobs that are not encrypted though, I don't see why it wouldn't be able to but ill run a manual test to make sure.

Reply:
I confirmed that it all works just fine if some files are encrypted and others are not, it is still able to index both of them.